shredstream.proto

```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+
+package shredstream;
+
+import "google/protobuf/empty.proto";
+import "shared.proto";
+
+message Heartbeat {
+  // don't trust IP:PORT from tcp header since it can be tampered over the wire
+  // this also prevents spamming an unwitting destination
+  shared.Socket socket = 1;
+  string auth_token = 2;
+}
+
+service Shredstream {
+  // RPC endpoint to send heartbeats to keep shreds flowing
+  rpc RefreshShredstreamSubscriber (stream Heartbeat) returns (google.protobuf.Empty) {}
```

Comment:
How about returning a response defined here so that if we do decide to add fields to the response we wouldn't break a bunch of clients, i.e. `HeartbeatResponse`

Reply:
Should heartbeat be a stream or can this be a unary method? Also thoughts on renaming to `SendHeartbeat`?

Reply:
Reason I say that is `Refresh` seems somewhat misleading since on initial requests clients aren't technically refreshing anything. Also having `Subscriber` in the method name seems like clients are modifying some resource named `Subscriber` on the server when in fact the client sending the request is the subscriber